### ISSUE TITLE
__rec_split size calculation off

### ISSUE THREAD

**Issue by agorrod** [wiredtiger/wiredtiger#1655]
I believe the size calculation in __rec_split is wrong if multiple splits fit onto a single page.

The code currently does:

```
if (inuse + r->split_size <= r->page_size)
           r->space_avail = r->split_size - WT_PAGE_HEADER_BYTE_SIZE(btree);
```

In this case we are attempting to fit all of the items on a single page - if we do that we only have a single page header written. The calculation should be: ```r->space_avail = r->split_size```. I believe that the code in ```__rec_split_fixup``` already deals with creating the page header from the boundary structure in the case where all of the content doesn't fit onto a single page.

**Comment by keithbostic** [comment]
I'm still working through this one, but for the record, I think the size calculation originally flagged by Alex was correct: if `split-size` is the size of the chunk we're supposed to push out if we split, it has to be corrected by `WT_PAGE_HEADER_BYTE_SIZE`.

For example, if `split-size` is 32KB, and the header is 100B, if we fill up 32KB of space, and then add 100B to it, the chunk we'll push out is larger than `split-size`. If we fill up (32KB - 100B), we can add 100B to it without going larger than `split-size`.

**Comment by keithbostic** [comment]
I agree with the second part of the change: if we're not taking a full split-sized chunk, there's no reason to take the page-header into account, and we might as well get those few extra bytes in case it lets us not split the page!

**Comment by keithbostic** [comment]
This one is complete, all of the changes have been merged (#1656, #1659).
